From 8ea251a06b31f73afd1324a71ff4f877ac6da2a4 Mon Sep 17 00:00:00 2001
From: Marouen Ghodhbane <marouen.ghodhbane@nxp.com>
Date: Thu, 18 Jan 2024 17:18:59 +0100
Subject: [PATCH 12/23] arm64: dts: imx8mp: add rpmsg and generic mbox dtsi

Signed-off-by: Marouen Ghodhbane <marouen.ghodhbane@nxp.com>
(add dma-coherent property and update generic mbox spi irqs)
Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
---
 .../dts/freescale/imx8mp-generic-mbox.dtsi    | 27 ++++++++++++++
 .../boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi | 37 +++++++++++++++++++
 2 files changed, 64 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-generic-mbox.dtsi
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-generic-mbox.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-generic-mbox.dtsi
new file mode 100644
index 000000000000..84ff983093ad
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-generic-mbox.dtsi
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2023 NXP
+ */
+
+/ {
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		gen_sw_mbox_reserved: gen-sw-mbox@fe000000 {
+			reg = <0 0xfe000000 0 0x1000>;
+			no-map;
+		};
+
+	};
+
+	gen_sw_mbox: generic-software-mailbox@fe000000 {
+		compatible = "fsl,generic-software-mbox";
+		reg = <0 0xfe000000 0 0x1000>;
+		#mbox-cells = <2>;
+		/* Use 2 unused SPI interrupts */
+		interrupts = <GIC_SPI 150 IRQ_TYPE_LEVEL_HIGH>, <GIC_SPI 151 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "irq", "remote_irq";
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi
new file mode 100644
index 000000000000..1bb2a8602379
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2022-2023 NXP
+ */
+
+#include "imx8mp-generic-mbox.dtsi"
+
+/ {
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		rpmsg_ca53_reserved: rpmsg-ca53@fe100000 {
+			reg = <0 0xfe100000 0 0x10000>;
+			no-map;
+		};
+
+		vdevbuffer_ca53: vdevbuffer-ca53@fe200000 {
+			compatible = "shared-dma-pool";
+			reg = <0 0xfe200000 0 0x100000>;
+			no-map;
+		};
+	};
+
+	rpmsg-ca53 {
+		compatible = "fsl,imx8mm-rpmsg";
+		reg = <0x0 0xfe100000 0x0 0x10000>; /* 64K for one rpmsg instance */
+		dma-coherent;
+		memory-region = <&vdevbuffer_ca53>;
+		mbox-names = "tx", "rx", "rxdb";
+		mboxes = <&gen_sw_mbox 0 0
+			  &gen_sw_mbox 1 0
+			  &gen_sw_mbox 2 0>;
+		vdev-nums = <1>;
+	};
+};
-- 
2.34.1

